File: app/src/main/java/vanderclay/comet/benson/franticsearch/API/CardSearchAPI.java
package vanderclay.comet.benson.franticsearch.API;

import android.content.ContentValues;
import android.content.Context;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;

import java.io.Serializable;
import java.util.List;
import io.magicthegathering.javasdk.api.CardAPI;
import io.magicthegathering.javasdk.resource.Card;


/**
 * Created by ben on 3/29/17.
 */

public class CardSearchAPI extends SQLiteOpenHelper implements Serializable {

    // Database Version
    private static final int DATABASE_VERSION = 1;

    // Database Name
    private static final String DATABASE_NAME = "frantic_search";

    // Contacts table name
    private static final String TABLE_CARDS = "Cards";

    // Columns name names
    private static final String id = "id";
    private static final String name = "name";
    private static final String manaCost = "mana_cost";
    private static final String cmc = "cmc";
    private static final String colors = "colors";
    private static final String type = "type";
    private static final String subtypes = "sub_types";
    private static final String rarity = "rarity";
    private static final String text = "text";
    private static final String power = "power";
    private static final String toughness = "toughness";
    private static final String loyalty = "loyalty";
    private static final String imageName = "image_name";
    private static final String reserved = "reserved";
    private static final String releaseDate = "release_date";
    private static final String starter = "starter";
    private static final String owned = "owned";

    //Constructor
    public CardSearchAPI(Context context) {
        super(context, DATABASE_NAME, null, DATABASE_VERSION);
    }


    // Creating Tables
    @Override
    public void onCreate(SQLiteDatabase db) {

        String CREATE_CONTACTS_TABLE = "CREATE TABLE " + TABLE_CARDS + "("
                + id + " INTEGER PRIMARY KEY, "
                + name + " TEXT, "
                + manaCost + " INTEGER, "
                + cmc + " INTEGER, "
                + colors + " TEXT, "
                + type + " TEXT, "
                + subtypes + " TEXT, "
                + rarity + " TEXT, "
                + text + " TEXT, "
                + power + " INTEGER, "
                + toughness + " INTEGER, "
                + loyalty + " INTEGER, "
                + imageName + " TEXT, "
                + reserved + " TEXT, "
                + releaseDate + " TEXT, "
                + starter + " TEXT, "
                + owned + " INTEGER "
                + ")";
        db.execSQL(CREATE_CONTACTS_TABLE);
    }

    // Upgrading database
    @Override
    public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
        // Drop older table if existed
        db.execSQL("DROP TABLE IF EXISTS " + DATABASE_NAME);
        // Create tables again
        onCreate(db);
    }

    /*
     * Get all the cards from the api we're using and creates an instance of the database.
     */
    public void addAllCards() {
        List<Card> allCards = CardAPI.getAllCards();
        SQLiteDatabase db = this.getWritableDatabase();
        for (Card apiCard : allCards) {
            System.out.println(apiCard);
            ContentValues values = new ContentValues();

//            values.put(KEY_NAME, contact.getName()); // Contact Name
//            values.put(KEY_PH_NO, contact.getPhoneNumber()); // Contact Phone Number
            values.put(id, apiCard.getId());
            values.put(name, apiCard.getName());
            values.put(manaCost, apiCard.getManaCost());
            values.put(cmc, apiCard.getCmc());
            values.put(colors, apiCard.getColors().toString());
            values.put(type, apiCard.getType());
            values.put(subtypes, apiCard.getSubtypes().toString());
            values.put(rarity, apiCard.getRarity());
            values.put(text, apiCard.getText());
            values.put(power, apiCard.getPower());
            values.put(toughness, apiCard.getToughness());
            values.put(loyalty, apiCard.getLoyalty());
            values.put(imageName, apiCard.getImageName());
            values.put(reserved, apiCard.isReserved());
            values.put(releaseDate, apiCard.getReleaseDate());
            values.put(starter, apiCard.isStarter());
            values.put(owned, 0);

            // Inserting Row
            db.insert(TABLE_CARDS, null, values);
            // Closing database connection
        }
        db.close();

    }
    //Get all Contacts
    public List<CardDO> getContacts() {
        List <CardDO> resultSet = null;

        SQLiteDatabase db = this.getReadableDatabase();
        Cursor cursor = db.rawQuery("SELECT * FROM "+TABLE_CARDS, null);

        if (cursor != null && cursor.moveToFirst()) {
            do{

                CardDO card = new CardDO();
                card.setId(cursor.getInt(0));
                card.setName(cursor.getString(1));
                card.setManaCost(cursor.getInt(2));
                card.setCmc(cursor.getInt(3));
                card.setColors(cursor.getString(4));
                card.setType(cursor.getString(5));
                card.setPower(cursor.getInt(6));
                card.setToughness(cursor.getInt(7));
                card.setLoyalty(cursor.getInt(8));
                card.setImageName(cursor.getString(9));
                card.setReserved(cursor.getInt(10) == 1);
                card.setReleaseDate(cursor.getString(11));
                card.setStarter(cursor.getString(12));
                card.setOwned(cursor.getInt(13) == 1);
                resultSet.add(card);
            }while(cursor.moveToFirst());
        }else{
            System.out.println("cursor is null or there's nothing in the database.");
        }
        return resultSet;
    }

    public static void main(String args[]){
//        CardSearchAPI cardSearchAPI = new CardSearchAPI();
//        cardSearchAPI.onCreate();
    }
}

File: app/src/main/java/vanderclay/comet/benson/franticsearch/API/CardSearchAPI.kt
package vanderclay.comet.benson.franticsearch.API

import android.content.ContentValues
import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

import java.io.Serializable
import io.magicthegathering.javasdk.api.CardAPI


class CardSearchAPI(context: Context) : SQLiteOpenHelper(context, CardSearchAPI.DATABASE_NAME, null, CardSearchAPI.DATABASE_VERSION), Serializable {


    // Creating Tables
    override fun onCreate(db: SQLiteDatabase) {

        val CREATE_CONTACTS_TABLE = """
            CREATE TABLE $TABLE_CARDS(
                $id INTEGER PRIMARY KEY,
                $name TEXT,
                $manaCost INTEGER,
                $cmc INTEGER,
                $colors TEXT,
                $type TEXT,
                $subtypes TEXT,
                $rarity TEXT,
                $text TEXT,
                $power INTEGER,
                $toughness INTEGER,
                $loyalty INTEGER,
                $imageName TEXT,
                $reserved TEXT,
                $releaseDate TEXT,
                $starter TEXT,
                $owned INTEGER
            )
        """
        db.execSQL(CREATE_CONTACTS_TABLE)
    }

    // Upgrading database
    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        // Drop older table if existed
        db.execSQL("DROP TABLE IF EXISTS " + DATABASE_NAME)
        // Create tables again
        onCreate(db)
    }

    /*
     * Get all the cards from the api we're using and creates an instance of the database.
     */
    fun addAllCards() {
        val allCards = CardAPI.getAllCards()
        val db = this.writableDatabase
        for (apiCard in allCards) {
            val values = ContentValues()

            values.put(id, apiCard.id)
            values.put(name, apiCard.name)
            values.put(manaCost, apiCard.manaCost)
            values.put(cmc, apiCard.cmc)
            values.put(colors, apiCard.colors.toString())
            values.put(type, apiCard.type)
            values.put(subtypes, apiCard.subtypes.toString())
            values.put(rarity, apiCard.rarity)
            values.put(text, apiCard.text)
            values.put(power, apiCard.power)
            values.put(toughness, apiCard.toughness)
            values.put(loyalty, apiCard.loyalty)
            values.put(imageName, apiCard.imageName)
            values.put(reserved, apiCard.isReserved)
            values.put(releaseDate, apiCard.releaseDate)
            values.put(starter, apiCard.isStarter)
            values.put(owned, 0)

            // Inserting Row
            db.insert(TABLE_CARDS, null, values)
            // Closing database connection
        }
        db.close()

    }

    val contacts: MutableList<CardDO>
        get() {
            val resultSet = mutableListOf<CardDO>()

            val db = this.readableDatabase
            val cursor = db.rawQuery("SELECT * FROM " + TABLE_CARDS, null)

            if (cursor != null && cursor.moveToFirst()) {
                do {
                    val card = CardDO(
                            cursor.getInt(0),
                            cursor.getString(1),
                            cursor.getString(2),
                            cursor.getInt(3),
                            cursor.getInt(4),
                            cursor.getString(5),
                            cursor.getString(6),
                            cursor.getInt(7),
                            cursor.getInt(8),
                            cursor.getInt(9),
                            cursor.getString(10),
                            cursor.getInt(11) == 1,
                            cursor.getString(12),
                            cursor.getString(13),
                            cursor.getInt(14) == 1
                    )
                    resultSet.add(card)
                } while (cursor.moveToFirst())
            } else {
                println("cursor is null or there's nothing in the database.")
            }
            return resultSet
        }

    companion object {

        // Database Version
        private val DATABASE_VERSION = 1

        // Database Name
        private val DATABASE_NAME = "frantic_search"

        // Contacts table name
        private val TABLE_CARDS = "Cards"

        // Columns name names
        private val id = "id"
        private val name = "name"
        private val manaCost = "mana_cost"
        private val cmc = "cmc"
        private val colors = "colors"
        private val type = "type"
        private val subtypes = "sub_types"
        private val rarity = "rarity"
        private val text = "text"
        private val power = "power"
        private val toughness = "toughness"
        private val loyalty = "loyalty"
        private val imageName = "image_name"
        private val reserved = "reserved"
        private val releaseDate = "release_date"
        private val starter = "starter"
        private val owned = "owned"

        @JvmStatic fun main(args: Array<String>) {
        }
    }
}
